sympy/matrices/eigen.py

```diff
@@ -372,6 +372,132 @@ def _is_diagonalizable(M, reals_only=False, **kwargs):
     return _is_diagonalizable_with_eigen(M, reals_only=reals_only)[0]
 
 
+#G&VL, Matrix Computations, Algo 5.4.2
+def _householder_vector(x):
+    if not x.cols == 1:
+        raise ValueError("Input must be a column matrix")
+    v = x.copy()
+    v_plus = x.copy()
+    v_minus = x.copy()
+    q = x[0, 0] / abs(x[0, 0])
+    norm_x = x.norm()
+    v_plus[0, 0] = x[0, 0] + q * norm_x
+    v_minus[0, 0] = x[0, 0] - q * norm_x
```

Comment:
I wonder where this method is from
I think that it's from https://github.com/vslobody/Householder-Bidiagonalization but it looks different than Golub & Van Loan's book. 
I know that the householder vector can be chosen in some arbitrary way. So it can be a trivial problem if this is correct.

Reply:
The original method which I had written was from  Golub & Van Loan's book, with some tweaks to remove redundant operations, but later I found out that the said method didn't work for complex matrices, so I looked for another method and found this one.